### PR TITLE
add reflectx.NonPointerNew to fix keyed list pointer element reflectx panic

### LINKED ISSUE #1161: crash in back button on sims/detector env/Test keyed list
### Describe the bug

another lovely reflectx issue :)

### How to reproduce

sims/ch2/detector, click on Env, then Test, then hit back button.

### Example code

_No response_

### Relevant output

```shell
cd /Users/oreilly/go/src/github.com/CompCogNeuro/sims/ch2/detector (from: {FileDirPath})
2024/08/20 13:52:50 ERROR Error: unable to set value 0x1400b49ae20 of type *env.Env (using underlying type: env.Env) from value 0x14008a95e78 of type *interface {} (using underlying type: *interface {}): not a supported type pair and direct assigning is not possible | cogentcore.org/core/core.bindMapValue[...].func2 /Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/keyedlist.go:284
2024/08/20 13:52:50 panic: runtime error: invalid memory address or nil pointer dereference
2024/08/20 13:52:50 
2024/08/20 13:52:50 ----- START OF STACK TRACE: -----
2024/08/20 13:52:50 goroutine 4 [running]:
runtime/debug.Stack()
	/opt/homebrew/Cellar/go/1.22.5/libexec/src/runtime/debug/stack.go:24 +0x64
cogentcore.org/core/system.HandleRecoverBase({0x1044538a0, 0x104e1bbb0})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/system/recover.go:50 +0x38
cogentcore.org/core/core.handleRecover({0x1044538a0, 0x104e1bbb0})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/recover.go:33 +0x48
cogentcore.org/core/core.(*renderWindow).eventLoop.func1()
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/renderwindow.go:368 +0x34
panic({0x1044538a0?, 0x104e1bbb0?})
	/opt/homebrew/Cellar/go/1.22.5/libexec/src/runtime/panic.go:770 +0x124
cogentcore.org/core/base/reflectx.SetRobust({0x1043c71c0, 0x14008a95e78}, {0x0, 0x0})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/base/reflectx/values.go:1064 +0xa78
cogentcore.org/core/core.bindMapValue[...].func1()
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/keyedlist.go:279 +0xec
cogentcore.org/core/core.(*WidgetBase).UpdateWidget(0x14008a95408)
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/update.go:23 +0x2c
cogentcore.org/core/core.(*WidgetBase).Update.func1({0x1059c8f10, 0x14008a95408}, 0x14000045330?)
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/update.go:61 +0x28
cogentcore.org/core/core.(*WidgetBase).Update.(*WidgetBase).WidgetWalkDown.func2({0x1059c8ec0, 0x14008a95408})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/widget.go:446 +0xa8
cogentcore.org/core/tree.(*NodeBase).WalkDown(0x14008a95408, 0x1400b49b6c0)
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/tree/nodebase.go:500 +0xf0
cogentcore.org/core/core.(*WidgetBase).WidgetWalkDown(...)
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/widget.go:444
cogentcore.org/core/core.(*WidgetBase).Update(0x14008a95408)
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/update.go:60 +0xbc
cogentcore.org/core/core.(*WidgetBase).UpdateChange(0x14008a95408, {0x0?, 0x0?, 0x0?})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/widgetevents.go:196 +0x40
cogentcore.org/core/core.openValueDialog.func1({0x10445be00?, 0x1400511bf50?})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/value.go:149 +0x2c
cogentcore.org/core/events.(*Listeners).Call(0x14000045538, {0x1045d8658, 0x1400b513960}, {0x14000045520, 0x1, 0x102fb2124?})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/events/listeners.go:61 +0x104
cogentcore.org/core/core.(*WidgetBase).HandleEvent.func1(0x1400511bf50)
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/widgetevents.go:244 +0x5c
cogentcore.org/core/base/tiered.(*Tiered[...]).Do(0x102d2a394, 0x140000455b8?)
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/base/tiered/tiered.go:30 +0x48
cogentcore.org/core/core.(*WidgetBase).HandleEvent(0x140088caa88, {0x1045d8658?, 0x1400b513960?})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/widgetevents.go:243 +0x124
cogentcore.org/core/core.(*Scene).Close.func1({0x1045d0fa0?, 0x140088caa88?}, 0x14000045768?)
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/scene.go:285 +0x44
cogentcore.org/core/core.(*Scene).Close.(*WidgetBase).WidgetWalkDown.func2({0x1045cad50, 0x140088caa88})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/widget.go:446 +0xa8
cogentcore.org/core/tree.(*NodeBase).WalkDown(0x140004ae388, 0x14008ab9d10)
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/tree/nodebase.go:500 +0xf0
cogentcore.org/core/core.(*WidgetBase).WidgetWalkDown(...)
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/widget.go:444
cogentcore.org/core/core.(*Scene).Close(0x140004ae388)
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/scene.go:284 +0xcc
cogentcore.org/core/core.makeStandardAppBar.func1.1({0x10445be00?, 0x14008974450?})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/app.go:108 +0x6c
cogentcore.org/core/events.(*Listeners).Call(0x14000045aa8, {0x1045d8658, 0x1400b5136c0}, {0x14000045a90, 0x1, 0x2?})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/events/listeners.go:61 +0x104
cogentcore.org/core/core.(*WidgetBase).HandleEvent.func1(0x14008974450)
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/widgetevents.go:244 +0x5c
cogentcore.org/core/base/tiered.(*Tiered[...]).Do(0x1b654670, 0x14000045b28?)
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/base/tiered/tiered.go:30 +0x48
cogentcore.org/core/core.(*WidgetBase).HandleEvent(0x1400897c008, {0x1045d8658?, 0x1400b5136c0?})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/widgetevents.go:243 +0x124
cogentcore.org/core/core.(*WidgetBase).Send(0x1400897c008, 0x5, {0x14000045d88?, 0x0?, 0x0?})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/widgetevents.go:180 +0x80
cogentcore.org/core/core.(*Events).handlePosEvent(0x140004aee80, {0x1045d88d8, 0x1400b4a4d20})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/events.go:471 +0x161c
cogentcore.org/core/core.(*Events).handleEvent(0x140004aee80, {0x1045d88d8, 0x1400b4a4d20})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/events.go:169 +0x58
cogentcore.org/core/core.(*Stage).mainHandleEvent(0x140038ed290, {0x1045d88d8, 0x1400b4a4d20})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/mainstage.go:389 +0x160
cogentcore.org/core/core.(*stages).mainHandleEvent(0x1400013e350, {0x1045d88d8, 0x1400b4a4d20})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/mainstage.go:397 +0x78
cogentcore.org/core/core.(*renderWindow).handleEvent(0x1400013e320, {0x1045d88d8, 0x1400b4a4d20})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/renderwindow.go:419 +0x1c8
cogentcore.org/core/core.(*renderWindow).eventLoop(0x1400013e320)
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/renderwindow.go:382 +0x90
created by cogentcore.org/core/core.(*renderWindow).goStartEventLoop in goroutine 1
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/renderwindow.go:353 +0x6c

2024/08/20 13:52:50 ----- END OF STACK TRACE -----
2024/08/20 13:52:50 SAVED CRASH LOG TO /Users/oreilly/Library/detector/crash-logs/crash_2024-08-20_13-52-50
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x98 pc=0x102eb45e8]

goroutine 4 [running]:
cogentcore.org/core/core.handleRecover({0x1044538a0, 0x104e1bbb0})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/recover.go:71 +0x218
cogentcore.org/core/core.(*renderWindow).eventLoop.func1()
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/renderwindow.go:368 +0x34
panic({0x1044538a0?, 0x104e1bbb0?})
	/opt/homebrew/Cellar/go/1.22.5/libexec/src/runtime/panic.go:770 +0x124
cogentcore.org/core/base/reflectx.SetRobust({0x1043c71c0, 0x14008a95e78}, {0x0, 0x0})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/base/reflectx/values.go:1064 +0xa78
cogentcore.org/core/core.bindMapValue[...].func1()
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/keyedlist.go:279 +0xec
cogentcore.org/core/core.(*WidgetBase).UpdateWidget(0x14008a95408)
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/update.go:23 +0x2c
cogentcore.org/core/core.(*WidgetBase).Update.func1({0x1059c8f10, 0x14008a95408}, 0x14000045330?)
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/update.go:61 +0x28
cogentcore.org/core/core.(*WidgetBase).Update.(*WidgetBase).WidgetWalkDown.func2({0x1059c8ec0, 0x14008a95408})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/widget.go:446 +0xa8
cogentcore.org/core/tree.(*NodeBase).WalkDown(0x14008a95408, 0x1400b49b6c0)
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/tree/nodebase.go:500 +0xf0
cogentcore.org/core/core.(*WidgetBase).WidgetWalkDown(...)
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/widget.go:444
cogentcore.org/core/core.(*WidgetBase).Update(0x14008a95408)
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/update.go:60 +0xbc
cogentcore.org/core/core.(*WidgetBase).UpdateChange(0x14008a95408, {0x0?, 0x0?, 0x0?})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/widgetevents.go:196 +0x40
cogentcore.org/core/core.openValueDialog.func1({0x10445be00?, 0x1400511bf50?})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/value.go:149 +0x2c
cogentcore.org/core/events.(*Listeners).Call(0x14000045538, {0x1045d8658, 0x1400b513960}, {0x14000045520, 0x1, 0x102fb2124?})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/events/listeners.go:61 +0x104
cogentcore.org/core/core.(*WidgetBase).HandleEvent.func1(0x1400511bf50)
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/widgetevents.go:244 +0x5c
cogentcore.org/core/base/tiered.(*Tiered[...]).Do(0x102d2a394, 0x140000455b8?)
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/base/tiered/tiered.go:30 +0x48
cogentcore.org/core/core.(*WidgetBase).HandleEvent(0x140088caa88, {0x1045d8658?, 0x1400b513960?})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/widgetevents.go:243 +0x124
cogentcore.org/core/core.(*Scene).Close.func1({0x1045d0fa0?, 0x140088caa88?}, 0x14000045768?)
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/scene.go:285 +0x44
cogentcore.org/core/core.(*Scene).Close.(*WidgetBase).WidgetWalkDown.func2({0x1045cad50, 0x140088caa88})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/widget.go:446 +0xa8
cogentcore.org/core/tree.(*NodeBase).WalkDown(0x140004ae388, 0x14008ab9d10)
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/tree/nodebase.go:500 +0xf0
cogentcore.org/core/core.(*WidgetBase).WidgetWalkDown(...)
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/widget.go:444
cogentcore.org/core/core.(*Scene).Close(0x140004ae388)
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/scene.go:284 +0xcc
cogentcore.org/core/core.makeStandardAppBar.func1.1({0x10445be00?, 0x14008974450?})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/app.go:108 +0x6c
cogentcore.org/core/events.(*Listeners).Call(0x14000045aa8, {0x1045d8658, 0x1400b5136c0}, {0x14000045a90, 0x1, 0x2?})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/events/listeners.go:61 +0x104
cogentcore.org/core/core.(*WidgetBase).HandleEvent.func1(0x14008974450)
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/widgetevents.go:244 +0x5c
cogentcore.org/core/base/tiered.(*Tiered[...]).Do(0x1b654670, 0x14000045b28?)
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/base/tiered/tiered.go:30 +0x48
cogentcore.org/core/core.(*WidgetBase).HandleEvent(0x1400897c008, {0x1045d8658?, 0x1400b5136c0?})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/widgetevents.go:243 +0x124
cogentcore.org/core/core.(*WidgetBase).Send(0x1400897c008, 0x5, {0x14000045d88?, 0x0?, 0x0?})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/widgetevents.go:180 +0x80
cogentcore.org/core/core.(*Events).handlePosEvent(0x140004aee80, {0x1045d88d8, 0x1400b4a4d20})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/events.go:471 +0x161c
cogentcore.org/core/core.(*Events).handleEvent(0x140004aee80, {0x1045d88d8, 0x1400b4a4d20})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/events.go:169 +0x58
cogentcore.org/core/core.(*Stage).mainHandleEvent(0x140038ed290, {0x1045d88d8, 0x1400b4a4d20})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/mainstage.go:389 +0x160
cogentcore.org/core/core.(*stages).mainHandleEvent(0x1400013e350, {0x1045d88d8, 0x1400b4a4d20})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/mainstage.go:397 +0x78
cogentcore.org/core/core.(*renderWindow).handleEvent(0x1400013e320, {0x1045d88d8, 0x1400b4a4d20})
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/renderwindow.go:419 +0x1c8
cogentcore.org/core/core.(*renderWindow).eventLoop(0x1400013e320)
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/renderwindow.go:382 +0x90
created by cogentcore.org/core/core.(*renderWindow).goStartEventLoop in goroutine 1
	/Users/oreilly/go/pkg/mod/cogentcore.org/core@v0.3.3-0.20240820000827-b7efbcd96c73/core/renderwindow.go:353 +0x6c

/Users/oreilly/go/src/github.com/CompCogNeuro/sims/ch2/detector/detector <b>failed</b> at: Tue Aug  20 13:52:52 PDT 2024 with error: exit status 2
```

### Platform

macOS

### PR DESCRIPTION
**Pull request by kkoreilly**
Also remove unused `reflectx.NewFrom`.

Fixes #1161

### CODE REVIEW

File: base/reflectx/pointers.go [hunk 1]
```diff
@@ -88,16 +88,22 @@ func UnderlyingPointer(v reflect.Value) reflect.Value {
...
-func NewFrom(v reflect.Value) reflect.Value {
-	u := Underlying(v)
-	if u.CanAddr() {
-		return u.Addr()
+// NonPointerNew has the same overall behavior as [reflect.New] except that
+// it traverses through any pointers such that a new zero non-pointer value
+// will be created in the end, so any pointers in the original type will not
+// be nil. For example, in pseudo-code, NonPointerNew(**int) will return
+// &(&(&(0))).
+func NonPointerNew(typ reflect.Type) reflect.Value {
```
Comment (rcoreilly) [inline base/reflectx/pointers.go]: NonNilPointerNew might be a better name?  It is not getting rid of the pointers, right?  just making them non-nil.

Comment (kkoreilly) [reply base/reflectx/pointers.go]: `NonNilPointerNew` still seems like a very awkward name; we should be able to come up with something better. I thought about `DeepNew` before (like `DeepEqual`), but that is somewhat unclear as well.

Comment (rcoreilly) [reply base/reflectx/pointers.go]: How about `SafePointerNew` or just `SafeNew` ?

Comment (kkoreilly) [reply base/reflectx/pointers.go]: Going with `NonNilNew`, which is both concise and clear.

**Review by rcoreilly — APPROVED** [review]

**Pull request merged**